test/e2e/dataplane/tcp_pod_to_service.go

```diff
@@ -11,37 +9,48 @@ import (
 
 var _ = Describe("[dataplane] Basic Pod to Service tests across clusters without discovery", func() {
 	f := framework.NewDefaultFramework("dataplane-p2s-nd")
-	It("Should be able to perform a Pod to Service TCP connection and exchange data between different clusters", func() {
-		testPod2ServiceTCP(f)
+	It("Should be able to perform a Pod to Service TCP connection and exchange data between different clusters NonGW to NonGW", func() {
+		testPod2ServiceTCP(f, framework.NonGatewayNode, framework.NonGatewayNode)
 	})
-})
 
-func testPod2ServiceTCP(f *framework.Framework) {
+	It("Should be able to perform a Pod to Service TCP connection and exchange data between different clusters NonGW to GW", func() {
```

Comment:
IMO, the following two tests should cover most of the use-cases.
1. GatewayNode to GatewayNode
2. nonGWNode to nonGWNode

The second use-case is like a superset and it covers most of the combinations.

So, please change this test (line 17) to Gateway to Gateway.

Also, make this as the first test to run and the above one (i.e., line 13) as the second test.